docs/source/models.rst

```diff
@@ -124,28 +132,45 @@ class:
+classification model trained on the `MNIST dataset <http://yann.lecun.com/exdb/mnist/>`_.
+The input has one named tensor where input sample is an image represented by a 28 × 28 × 1 array
+of float32 numbers. The output is an unnamed tensor that has 10 units specifying the
+likelihood corresponding to each of the 10 classes. Note that the first dimension of the input
+and the output is the batch size and is thus set to -1 to allow for variable batch sizes. 
+
+.. code-block:: yaml
+
+  signature:
+      inputs: '[{"name": "images", "dtype": "uint8", "shape": [-1, 28, 28, 1]}]'
+      outputs: '[{"shape": [-1, 10], "dtype": "float32"}]'
 
 Signature Enforcement
 ~~~~~~~~~~~~~~~~~~~~~
-When scoring a model that includes a signature, inputs are validated based on the signature's input
-schema. This input schema enforcement checks input column ordering and column types, raising an
-exception if the input is not compatible. This enforcement is applied in MLflow before calling the
-underlying model implementation. Note that this enforcement only applies when using :ref:`MLflow
+Schema enforcement checks the provided input against the model's signature
+and raises an exception if the input is not compatible. This enforcement is applied in MLflow before
+calling the underlying model implementation. Note that this enforcement only applies when using :ref:`MLflow
 model deployment tools <built-in-deployment>` or when loading models as ``python_function``. In
 particular, it is not applied to models that are loaded in their native format (e.g. by calling
 :py:func:`mlflow.sklearn.load_model() <mlflow.sklearn.load_model>`).
 
-Column Ordering Enforcement
-"""""""""""""""""""""""""""
-The input columns are checked against the model signature. If there are any missing columns,
-MLflow will raise an exception. Extra columns that were not declared in the signature will be
-ignored. If the input schema in the signature defines column names, column matching is done by name
-and the columns are reordered to match the signature. If the input schema does not have column
-names, matching is done by position (i.e. MLflow will only check the number of columns).
+Name Ordering Enforcement
+"""""""""""""""""""""""""
+The input names are checked against the model signature. If there are any missing inputs,
+MLflow will raise an exception. Extra inputs that were not declared in the signature will be
+ignored. If the input schema in the signature defines input names, input matching is done by name
+and the inputs are reordered to match the signature. If the input schema does not have input
+names, matching is done by position (i.e. MLflow will only check the number of inputs).
 
-Column Type Enforcement
+Input Type Enforcement
 """""""""""""""""""""""
-The input column types are checked against the signature. MLflow will perform safe type conversions
+The input types are checked against the signature. MLflow will perform safe type conversions
```

Comment:
We should clarify that we only do type conversions for dataframe-like schema. tensor type checking is strict.